pkg/utils/transform/transform_test.go

```diff
@@ -171,3 +168,172 @@ func assertOnResourceList(t *testing.T, items []unstructured.Unstructured, key,
 		}
 	}
 }
+
+func TestReplaceImages(t *testing.T) {
+	t.Run("ignore_non_deployment", func(t *testing.T) {
+		testData := path.Join("testdata", "test-replace-kind.yaml")
+		expected, _ := mf.NewManifest(testData, true, nil)
+
+		manifest, err := mf.NewManifest(testData, true, nil)
+		assetNoEror(t, err)
+		err = manifest.Transform(transform.ReplaceDeploymentImages("IMAGE_"))
+
+		assetNoEror(t, err)
+		assertEqual(t, manifest.Resources, expected.Resources)
+	})
+
+	t.Run("of_containers", func(t *testing.T) {
+		image := "foo.bar/image/controller"
+		os.Setenv("IMAGE_PIPELINE_CONTROLLER", image)
+		testData := path.Join("testdata", "test-replace-image.yaml")
+
+		manifest, err := mf.NewManifest(testData, true, nil)
+		assetNoEror(t, err)
+		err = manifest.Transform(transform.ReplaceDeploymentImages("IMAGE_PIPELINE_"))
+
+		assetNoEror(t, err)
+		assertDeployContainersHasImage(t, manifest.Resources, "controller", image)
+		assertDeployContainersHasImage(t, manifest.Resources, "sidecar", "busybox")
+	})
+
+	t.Run("of_container_args", func(t *testing.T) {
+		image := "foo.bar/image/bash"
+		os.Setenv("IMAGE_PIPELINE_-BASH", image)
+		testData := path.Join("testdata", "test-replace-image.yaml")
+
+		manifest, err := mf.NewManifest(testData, true, nil)
+		assetNoEror(t, err)
+		err = manifest.Transform(transform.ReplaceDeploymentImages("IMAGE_PIPELINE_"))
+
+		assetNoEror(t, err)
+		assertDeployContainerArgsHasImage(t, manifest.Resources, "-bash", image)
+		assertDeployContainerArgsHasImage(t, manifest.Resources, "-git", "git")
+	})
+
+	t.Run("of_task_addons", func(t *testing.T) {
+		image := "foo.bar/image/buildah"
+		os.Setenv("IMAGE_ADDONS_PARAM_BUILDER_IMAGE", image)
+		os.Setenv("IMAGE_ADDONS_push", image)
+		testData := path.Join("testdata", "test-replace-addon-image.yaml")
+
+		manifest, err := mf.NewManifest(testData, true, nil)
+		assetNoEror(t, err)
+		err = manifest.Transform(transform.ReplaceTaskImage("IMAGE_ADDONS_", "PARAM_"))
+
+		assetNoEror(t, err)
+		assertTaskImage(t, manifest.Resources, "push", image)
+		assertParamHasImage(t, manifest.Resources, "BUILDER_IMAGE", image)
+	})
+}
+
+func assetNoEror(t *testing.T, err error) {
```

Comment:
typo: `assert`

Reply:
may be we can move all the assert functions in to a separate file in `pkg/utils/<something like test helpers>`

Reply:
Its better to address in the another PR https://github.com/openshift/tektoncd-pipeline-operator/issues/263